implement full jdbc crud functionality for book entity

Mwxyz2022 opened the pull request:
- added mysql jdbc driver dependency to pom.xml
- created init_db.sql script for the books table schema
- implemented book model and custom data processing exception
- developed a connection utility to manage db connections via a properties file
- defined the bookdao interface with standard CRUD operations
- implemented the bookdao interface using preparedstatement for all operations
- used try-with-resources for reliable resource management
- added a helper method to parse book objects from resultset to reduce duplication
- implemented robust error handling and checks for affected rows
- created a main method to test and demonstrate all crud functionalities

Mwxyz2022 committed:
implement full jdbc crud functionality for book entity

- added mysql jdbc driver dependency to pom.xml
- created init_db.sql script for the books table schema
- implemented book model and custom data processing exception
- developed a connection utility to manage db connections via a properties file
- defined the bookdao interface with standard CRUD operations
- implemented the bookdao interface using preparedstatement for all operations
- used try-with-resources for reliable resource management
- added a helper method to parse book objects from resultset to reduce duplication
- implemented robust error handling and checks for affected rows
- created a main method to test and demonstrate all crud functionalities

mateacademy-ai-mentor commented on src/main/java/mate/academy/Main.java:
```diff
@@ -1,7 +1,68 @@
+            System.out.println("Book with id" + bookId + " not found.");
+        }
+
+        Book bookToUpdate = foundBookOptional.get();
```
Issue: You are calling foundBookOptional.get() without checking if the Optional contains a value. If the book is not found, this will throw a NoSuchElementException. Please ensure you only call get() on an Optional after checking isPresent() or use orElseThrow().

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your hard work! Your implementation meets all the core requirements and is well-structured. However, there is one critical issue that needs to be addressed before approval: in your `Main.java`, you are calling `foundBookOptional.get()` without checking if the value is present. This can cause a `NoSuchElementException` if the book is not found, which is unsafe and violates best practices for using `Optional`. Please update your code to check if the value is present (using `isPresent()`, `ifPresent()`, or `orElseThrow()`) before accessing it. Once this is fixed, your solution will be ready for approval! 😊 Keep up the great work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

Mwxyz2022 committed:
prevent exception on book update

- moved the book update logic inside the isPresent() check
- prevented a potential NoSuchElementException when the book is not found

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 Your implementation meets all the core requirements and passes the checklist. All previous issues have been addressed, and your code is robust, well-structured, and ready to be merged. Keep up the excellent work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
